### DISCUSSION TITLE
to verify my very first run results

### DISCUSSION THREAD

**Discussion by chrisalexl2020** [MoonriseSoftwareCalifornia/AspNetCore.Identity.CosmosDb#11] [Q&A] [answered]
Hi Eric,
Thank you for your work.
I think I followed all the instructions for installation. I use local Cosmos DB Emulator, 7 containers has been created: Identity, Identity_DeviceFlowCodes, Identity_Logins, Identity_PersistedGrant, Identity_Roles, Identity_Tokens, Identity_UserRoles, and they are not same as the tables created if it is with SQL server. 
After one user registration, I see record created in Identity, Identity_Tokens only, not records in other containers be added, no Role, no UserRoles.

Email verification succeed.

Question:
1: Anything wrong I have done?
2: is it a partial implementation for ASP.Net Core Identity?

Your help will be greatly appreciated. Thanks.

Chris

**Comment by toiyabe** [comment, marked as answer]
Hello @chrisalexl2020, 
First, I apologize for the late reply!
You did everything correctly. You are right the container names are different than the table names in SQL server. That is OK.
Cosmos is a NoSQL database so it doesn't have "tables." The closest thing perhaps are "containers."

SQL tables are highly structured. All rows are required to have the same schema.

NoSQL tables, however, can have records with different schemas. The records will have an "identifier" that marks entities with similar schemas together.

The result is that a single Cosmos database can, have a single container. And within that container it can have entities of various different schema.

In my implementation of this framework I decided to lump more than one entity type in a single container.

Why split or lump? It's a matter of performance and convenience.

I suppose I could have had everything in a single container.

Well, I'm not an expert with Cosmos DB so I'm always open to ideas on how to improve this package.

Thanks! And again I'm so sorry for the late reply!!
